### PR TITLE
Feature/edit state for cards

### PR DESCRIPTION
**Pull request by delete-44**
- [x] Adds bio field to members
- [x] Adds editable state for member cards
- [x] Adds API route to update members with "strong params" equivalent, to only permit bio (for now))

https://github.com/orbit-love/member-directory/assets/45462299/13993a8e-71b9-470e-b9bc-a6e03eb80c72

### CODE REVIEW

File: components/member-card-editable.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import Bio from "./bio";
+import { useState } from "react";
+
+export default function MemberCardEditable({ member }) {
+  const [editState, setEditState] = useState(false);
+
+  return (
+    <li className="flex flex-col gap-6 xl:flex-row" key={member.id}>
+      {/* We cannot use next/Image here since avatars are remotely loaded
+          & we do not know the full list of sources for them */}
```
Comment (phacks) [inline components/member-card-editable.js]: Not sure I understand this part: “we do not know the full list of sources for them”

File: pages/api/update-member.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  }
+
+  try {
+    // Attempt to update the user
+    await updateMember(req.body);
+
+    res.status(200).send();
+  } catch (error) {
+    // If an error occurred, send back a 500 status and the error message
+    res.status(500).send({ message: `Error updating user: ${error.message}` });
```
Comment (phacks) [inline pages/api/update-member.js:16]: You might want to log the error server-side too (if that’s possible in Vercel?)

File: components/member-card-editable.js (hunk 1 above)
Comment (delete-44) [reply components/member-card-editable.js]: Hmmmm yeah I'll add some more context to this, thanks!

File: components/member-card-edit-state.js [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
+      body: JSON.stringify({ id: member.id, bio }),
+    });
+
+    if (response.ok) {
+      // On successful response, set bio to new value & return to default view
+      member.bio = bio;
+      setEditState(false);
+    } else {
+      // On error....
+      console.error(`Error: ${response.status}`);
```
[the remark this answers is not in the document]
Comment (delete-44) [reply components/member-card-edit-state.js]: Bit of a useless screenshot, but just wanted to confirm that the logs are showing in the server but not shown to the forntend ^^ 
image

File: components/member-card.js [hunk 4]
```diff
@@ -7,8 +7,11 @@ export default function MemberCard({ member, editable }) {
 
   return (
     <li className="flex flex-col gap-6 xl:flex-row" key={member.id}>
-      {/* We cannot use next/Image here since avatars are remotely loaded
-          & we do not know the full list of sources for them */}
+      {/* We cannot use next/Image here since avatars are remotely loaded.
+          Loading remote images with next requires you to list valid URL
+          patterns in your next.config.js, and we do not control the source
+          of member profile image URLs.
```
Comment (phacks) [inline components/member-card.js:13]: Not for this PR, but I think we could work around this by casting a wide net: we look at common URL patterns in the production database for avatar urls, and we add them to the config.

**Review by phacks — APPROVED** [review]

**Pull request merged**